owncloud/core
# Allow core to whitelist apps for a user

This changeset allows core to whitelist apps for
a user. This way core gets controll over the apps
which a user can access or not. But this data needs
to be provided either by the app or the caller of
specific method in the IRegulateAppAcces to set.

Signed-off-by: Sujith H <sharidasan@owncloud.com>



## Description

In this PR, I have tried to grab the information about access of apps for a user. The idea here is to have the data available for core. So lets say for example the SettingsManager resides in core. And SettingsManager cannot know what are the apps whitelisted for a user, unless the data is shared with core. So any app(s) which requires to share this data to core, should use this API. So that core has this information.

The data is saved in the array where user id is the key and the apps are the list. So we get the details per user.

## Related Issue




- Fixes <issue_link>

## Motivation and Context

The whitelisted apps details should be available to core. 

## How Has This Been Tested?



- Enable guests and customgroups app
- Create a guest user
- Now login as guest user and navigate to personal settings page
- customgroups ( by default ) will not appear in the list.

## Screenshots (if appropriate):

## Types of changes


## Checklist:
